source/grammar/qasm3.g4

```diff
@@ -447,27 +455,22 @@ kernelDeclaration
     : 'kernel' Identifier ( LPAREN classicalTypeList? RPAREN )? returnSignature? SEMICOLON
     ;
 
-// if have kernel w/ out args, is ambiguous; may get matched as identifier
-kernelCall
-    : Identifier LPAREN expressionList? RPAREN
+// if have function call w/ out args, is ambiguous; may get matched as identifier
+kernelOrSubroutineCall
+    : Identifier ( LPAREN expressionList? RPAREN )? indexIdentifierList?
```

Comment:
I'm not sure this is correct. I think we need a list which allows either expression or index identifier and put that list into the parens. Something like `exprAndIndexIdList`

Reply:
The test to add would be something like `sub_call(a, b, q[0] || q[1])` and see if that works...

Reply:
This was wrong... As @stevenheidel mentioned, `qubit` arguments must be in parenthesis in any calls. 

```
kernelOrSubroutineCall
    : Identifier ( LPAREN expressionList? RPAREN )?
    ;
```

Then, `sub_call(a, b, q[0] || q[1])` is parsed as

```
kernelOrSubroutineCall
  sub_call
  (
  expressionList
    expression
      expressionTerminator
        a
    ,
    expression
      expressionTerminator
        b
    ,
    expression
      expression
        expressionTerminator
          expressionTerminator
            q
          [
          expression
            expressionTerminator
              0
          ]
      ||
      logicalAndExpression
        bitOrExpression
          xOrExpression
            bitAndExpression
              equalityExpression
                comparisonExpression
                  bitShiftExpression
                    additiveExpression
                      multiplicativeExpression
                        expressionTerminator
                          expressionTerminator
                            q
                          [
                          expression
                            expressionTerminator
                              1
                          ]
```

Reply:
Right--that's why I am suggesting a list that can either include expressions or index identifiers, which would be in the parenthesis. Something like:
```
exprOrIndexId : expression | indexIdentifier;
exprOrIndexIdList : (exprOrIndexId COMMA)* exprOrIndexId
```

Reply:
I'm not sure we should add `||`, `:` to expression as they are specific expressions only allowed for registers. If we added to expression you could do `int x = [2:3]` which clearly doesn't make sense.